signer/tests/integration/withdrawal_reject.rs

```diff
@@ -603,3 +604,135 @@ async fn reject_withdrawal_validation_request_being_fulfilled() {
+    fetch_canonical_bitcoin_blockchain(&db, rpc).await;
+
+    // Different: we need to store a row in the dkg_shares table so that we
+    // have a record of the scriptPubKey that the signers control. We need
+    // this so that the donation gets picked up correctly below.
+    setup.store_dkg_shares(&db).await;
+
+    // Normal: The signers normally have a UTXO, so we add one here too. It
+    // is necessary when checking for whether the withdrawal being
+    // fulfilled by a sweep transaction that is in the mempool.
+    setup.store_donation(&db).await;
+
+    // Normal: the request and how the signers voted needs to be added to
+    // the database. Here the bitmap in the withdrawal request object
+    // corresponds to how the signers voted.
+    setup.store_withdrawal_requests(&db).await;
+    setup.store_withdrawal_decisions(&db).await;
+
+    // Normal: We do not reject a withdrawal requests until more than
+    // WITHDRAWAL_BLOCKS_EXPIRY blocks have been observed since the smart
+    // contract that created the withdrawal request has bene observed.
+    faucet.generate_blocks(WITHDRAWAL_BLOCKS_EXPIRY + 1);
+
+    // Normal: the signer follows the bitcoin blockchain and event observer
+    // should be getting new block events from bitcoin-core. We haven't
+    // hooked up our block observer, so we need to manually update the
+    // database with new bitcoin block headers.
+    fetch_canonical_bitcoin_blockchain(&db, rpc).await;
+
+    // Different: We submit a sweep transaction into the mempool so that
+    // the TestSweepSetup2 struct has the sweep_tx_info set. We also need
+    // to submit the transaction in order for
+    // `TestSweepSetup2::store_bitcoin_withdrawals_outputs` to work as
+    // expected.
+    setup.broadcast_sweep_tx(rpc);
+
+    // Different: We're adding a row that let the signer know that someone
+    // may have tried to fulfill the withdrawal request. If that
+    // transaction is spending the current signer UTXO, then it could
+    // possibly be in the mempool. Since the signers' UTXO is a donation,
+    // we're saying that the coordinator may have tried to fulfill the
+    // withdrawal.
+    setup.store_bitcoin_withdrawals_outputs(&db).await;
+
+    // Generate the transaction and corresponding request context.
+    let (reject_withdrawal_tx, req_ctx) = make_withdrawal_reject(&setup, &db).await;
+
+    let validation_result = reject_withdrawal_tx.validate(&ctx, &req_ctx).await;
+    match validation_result.unwrap_err() {
+        Error::WithdrawalRejectValidation(ref err) => {
+            assert_eq!(err.error, WithdrawalRejectErrorMsg::RequestStillActive)
+        }
+        err => panic!("unexpected error during validation {err}"),
+    }
+
+    let withdrawals = setup.withdrawals.drain(..).collect::<Vec<_>>();
+    setup.submit_sweep_tx(rpc, faucet);
+
+    faucet.generate_blocks(WITHDRAWAL_MIN_CONFIRMATIONS);
+
+    setup.store_sweep_tx(&db).await;
```

Comment:
nit: can we move this `store_sweep_tx` right after `submit_sweep_tx`? Otherwise it may seems we need to have the blocks in the middle

Reply:
Updated in https://github.com/stacks-network/sbtc/pull/1411/commits/856e346f565eb336631484851d2785208fe2b9b0